triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala

```diff
@@ -29,11 +29,23 @@ import com.daml.platform.sandbox.SandboxServer
 import com.daml.platform.sandbox.config.SandboxConfig
 import com.daml.platform.services.time.TimeProviderType
 import com.daml.ports.Port
-
-import scala.concurrent.{ExecutionContext, Future}
+import com.daml.bazeltools.BazelRunfiles
+import scala.concurrent._
+import scala.sys.process.Process
+import eu.rekawek.toxiproxy._
 
 object TriggerServiceFixture {
 
+  // Might throw IOException (unlikely). Best effort. There's a small
+  // chance that having found one, it gets taken before we get to use
+  // it.
+  private def findFreePort(): Int = {
+    val socket = new java.net.ServerSocket(0)
+    val port = socket.getLocalPort()
+    socket.close()
+    port
```

Comment:
You probably need to call `close()` in a `finally` block.

```suggestion
    val socket = new java.net.ServerSocket(0)
    try {
      socket.getLocalPort
    } finally {
      socket.close()
    }
```